rest-admin/src/test/java/pl/edu/icm/unity/restadm/TestGroupsManagement.java

```diff
@@ -60,6 +61,25 @@ public void addedGroupIsReturned() throws Exception
 		assertThat(groupContent.getSubGroups().contains("/subgroup"), is(true));
 	}
 
+	@Test
+	public void recursivelyAddedGroupIsReturned() throws Exception
```

Comment:
Well, we can have an integration test here (optionally really), but a must have is to test the engine operation. So tests like this are needed in engine. And perhaps not one but at least two (creating all parents, or few parents).

Reply:
done